runtime/lua/vim/treesitter.lua

```diff
@@ -184,9 +184,9 @@ function Query:match_preds(match, pattern, bufnr)
       if start_row ~= end_row then
         return false
       end
-      return regexes[i]:match_line(bufnr, start_row, start_col, end_col)
-    else
-      return false
+      if not regexes[i]:match_line(bufnr, start_row, start_col, end_col) then
+        return false
+      end
```

Comment:
would prefer to keep the explicit `else`, perhaps with a hint

    else  -- Unknown predicate always matches. #12173
      return true

Reply:
The thing is that we don't want to return here to go through all the predicates in this match. This returning `true` only at the very end of the function.

Reply:
ok that's definitely worth calling out in a comment :)

Reply:
You are right, it should be solved in the last push.